react-custom-hooks/client/src/components/useTodos.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+
+/**
+ * Manages the Todos by reading from and writing to the backend API.
+ *
+ * @returns {Object} with the following properties:
+ *   isLoading: a boolean indicating if the data has not yet loaded;
+ *   error: any error that occurred
+ *   todos: a list of To Do items;
+ *   addTodo: a function (newTodo) that will add a new To Do;
+ *   toggleCompleted: a function (todoId) that will toggle the completion of an existing To Do.
+ */
+export function useTodos() {
+  const [todos, setTodos] = useState();
+  const [error, setError] = useState();
+  useEffect(() => {
+    /* TODO: If `todos` hasn't been defined yet, read the items from the API
+     * and set them into the `todos` state.
+     */
+    async function getTodos() {
+      const data = await readTodos();
```

Comment:
Every time you use `await` you need to make sure it is wrapped in a `try/catch` (either within the function where you use it or up in the call stack). None of your `await` usages are in try/catch.